src/UhdmAst.cpp

```diff
@@ -2209,11 +2211,28 @@ namespace UhdmAst {
                   fromp,
                   ((AstSelExtract*)item)->msbp()->cloneTree(true),
                   ((AstSelExtract*)item)->lsbp()->cloneTree(true));
+	      std::cout << "AstSelExtract\n";
+	    } else if(item->type() == AstType::en::atConst) {
+	      std::cout << "AstSelPlus2\n";
+              //select = new AstSelPlus(new FileLine("uhdm"), fromp, bitp, nullptr);
+              select = new AstSelBit(new FileLine("uhdm"), fromp, bitp);
+		select->dumpTree(cout, "select");
+		fromp->dumpTree(cout, "fromp");
+		bitp->dumpTree(cout, "bitp");
+            } else if (item->type() == AstType::atSelPlus) {
+	      std::cout << "AstSelBit3\n";
+		select->dumpTree(cout, "select");
+                select = new AstSelPlus(new FileLine("uhdm"), fromp, bitp->op2p(), bitp->op3p());
```

Comment:
I am not sure if I read the code correctly, but could we do sth like the following?
```
AstSelPlus* selplusp = VN_CAST(item, SelPlus);
select = new AstSelPlus(new FileLine("uhdm"), fromp, selplusp->bitp(), selplusp->widthp());
```

This way we avoid using `op`s directly

Reply:
Or maybe we could even avoid generating the internal `AstSelPlus` node if we don't use it anyway? Or is it needed in some other cases?

Reply:
Now that I think about it, maybe we don't even need to create new nodes here.
We could probably just do sth like:
```
AstSelPlus* selplusp = VN_CAST(item, SelPlus);
selplusp->fromp(fromp);
select = selplusp;
```
???

Reply:
It looks like your first suggestion works well. I'd stick to it.